--- frrcommon.sh.in	2019-03-01 07:32:04.000000000 +0800
+++ frrcommon.sh.in	2019-03-17 12:26:57.026113465 +0800
@@ -270,7 +270,7 @@
 	[ -s "$oldcfg" ] || return 0
 	grep -v '^[[:blank:]]*\(#\|$\)' "$oldcfg" > /dev/null || return 0
 
-	log_warning_msg "Reading deprecated $oldcfg.  Please move its settings to $C_PATH/daemons and remove it."
+	log_warning_msg "Reading deprecated $oldcfg.  Please move its settings to $C_PATH/daemons.conf and remove it."
 
 	# save off settings from daemons for the OR below
 	for dmn in $DAEMONS; do eval "_new_$dmn=\${$dmn:-no}"; done
@@ -281,13 +281,13 @@
 	for dmn in $DAEMONS; do eval "test \$_new_$dmn != no && $dmn=\$_new_$dmn; unset _new_$dmn"; done
 }
 
-[ -r "$C_PATH/daemons" ] || {
+[ -r "$C_PATH/daemons.conf" ] || {
 	log_failure_msg "cannot run $@: $C_PATH/daemons does not exist"
 	exit 1
 }
-. "$C_PATH/daemons"
+. "$C_PATH/daemons.conf"
 
-load_old_config "$C_PATH/daemons.conf"
+load_old_config "$C_PATH/daemons"
 load_old_config "/etc/default/frr"
 load_old_config "/etc/sysconfig/frr"
 
--- frr.in	2019-03-01 07:32:04.000000000 +0800
+++ frr.in.new	2019-03-17 13:19:33.821508061 +0800
@@ -500,11 +500,11 @@
 #########################################################
 
 # Config broken but script must exit silently.
-[ ! -r "$C_PATH/daemons" ] && exit 0
+[ ! -r "$C_PATH/daemons.conf" ] && exit 0
 
 # Load configuration
-. "$C_PATH/daemons"
 . "$C_PATH/daemons.conf"
+[ -r "$C_PATH/daemons" ] && . "$C_PATH/daemons"
 
 # Read configuration variable file if it is present
 [ -r /etc/default/frr ] && . /etc/default/frr
